test/e2e/user_workloads_test.go

```diff
@@ -0,0 +1,116 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	"github.com/codeready-toolchain/toolchain-e2e/wait"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestUserWorkloads(t *testing.T) {
+	suite.Run(t, &userWorkloadsTestSuite{})
+}
+
+type userWorkloadsTestSuite struct {
+	baseUserIntegrationTest
+}
+
+func (s *userWorkloadsTestSuite) SetupSuite() {
+	s.ctx, s.hostAwait, s.memberAwait = WaitForDeployments(s.T(), &v1alpha1.UserSignupList{})
+	s.setApprovalPolicyConfig("automatic")
+}
+
+func (s *userWorkloadsTestSuite) TearDownTest() {
+	s.ctx.Cleanup()
+}
+
+func (s *userWorkloadsTestSuite) TestIdler() {
+	// Provision a user to idle with a short idling timeout
+	s.createAndCheckUserSignup(true, "test-idler", "test-idler@redhat.com", ApprovedByAdmin()...)
+	idler, err := s.memberAwait.WaitForIdler("test-idler-dev")
+	require.NoError(s.T(), err)
+
+	// Noise
+	idlerNoise, err := s.memberAwait.WaitForIdler("test-idler-stage")
+	require.NoError(s.T(), err)
+
+	// Create payloads for both users
+	podsToIdle := s.prepareWorkloads(idler)
+	podsNoise := s.prepareWorkloads(idlerNoise)
+
+	// Set a short timeout for one of the idler to trigger pod idling
+	idler.Spec.TimeoutSeconds = 3
+	err = s.memberAwait.Client.Update(context.TODO(), idler)
+	require.NoError(s.T(), err)
+
+	// Wait for the pods to be deleted
+	for _, p := range podsToIdle {
+		err := s.memberAwait.WaitUntilPodDeleted(p.Namespace, p.Name)
+		require.NoError(s.T(), err)
+	}
+
+	// Noise pods are still there
+	labels := make(map[string]string)
+	labels["app"] = "idler-hello-openshift"
+	_, err = s.memberAwait.WaitForPods(idlerNoise.Name, labels, len(podsNoise), wait.UntilPodRunning())
+	require.NoError(s.T(), err)
+}
+
+func (s *userWorkloadsTestSuite) prepareWorkloads(idler *v1alpha1.Idler) []*corev1.Pod {
```

Comment:
```suggestion
func (s *userWorkloadsTestSuite) prepareDeploymentWorkloads(idler *v1alpha1.Idler) []*corev1.Pod {
```

Reply:
so there is room for other kinds of workloads (replicaset, standalone pods, etc.)